language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/diagnostic/DiagnosticsHelper.java

```diff
@@ -190,7 +191,8 @@ public Map<String, List<Diagnostic>> getLatestDiagnostics(DocumentServiceContext
         if (project.get().kind() == ProjectKind.SINGLE_FILE_PROJECT) {
             projectRoot = projectRoot.getParent();
         }
-        PackageCompilation compilation = workspace.waitAndGetPackageCompilation(context.filePath()).orElseThrow();
+        PackageCompilation compilation = workspace.waitAndGetPackageCompilation(context.filePath(),
+                context.operation() == LSContextOperation.TXT_DID_CHANGE).orElseThrow();
```

Comment:
Since the probability of reaching `orElseThrow()` is high now, we should check if the returned compilation is empty and, if empty, just return the empty map.